io/post_nems_routines.F90

```diff
@@ -100,22 +126,74 @@ subroutine post_alctvars(imi,jmi,lmi,mype,nwtlpes,lead_write, mpicomp,  &
 !
 !     counts, disps for gatherv and scatterv
 !
+      isumm=0
+      isumm2=0
+      allocate(isxa(0:num_procs-1) )
+      allocate(jsxa(0:num_procs-1) )
+      allocate(iexa(0:num_procs-1) )
+      allocate(jexa(0:num_procs-1) )
+
       do i = 1, num_procs
-       icnt(i-1) = (jtegrp(i)-jtsgrp(i)+1)*im
-       idsp(i-1) = (jtsgrp(i)-1)*im
+!       icnt(i-1) = (jtegrp(i)-jtsgrp(i)+1)*im
+!       idsp(i-1) = (jtsgrp(i)-1)*im
 !       if ( mype .eq. lead_write ) then
 !           print *, ' i, icnt(i),idsp(i) = ',i-1,icnt(i-1),idsp(i-1)
 !       end if
+       icnt(i-1) = (jtegrp(i)-jtsgrp(i)+1)*(itegrp(i)-itsgrp(i)+1)
+       isxa(i-1) = itsgrp(i)
+       iexa(i-1) = itegrp(i)
+       jsxa(i-1) = jtsgrp(i)
+       jexa(i-1) = jtegrp(i)
+       idsp(i-1) = isumm
+       isumm=isumm+icnt(i-1)
+       if(jtsgrp(i)==1 .or. jtegrp(i)==jm) then
+          icnt2(i-1) = (itegrp(i)-itsgrp(i)+1)
+       else
+          icnt2(i-1) = 0
+       endif
+       idsp2(i-1)=isumm2
+       if(jtsgrp(i)==1 .or. jtegrp(i)==jm) isumm2=isumm2+(itegrp(i)-itsgrp(i)+1)
+
       enddo
+      write(6,'(a25,i4,16i8)') 'JESSE,me,icnt2,idsp2=',me,icnt2(0:num_procs-1),idsp2(0:num_procs-1)
 !
 !     extraction limits -- set to two rows
 !
       jsta_2l = max(jsta - 2,  1 )
       jend_2u = min(jend + 2, jm )
+      if(modelname=='GFS') then
+        ista_2l=max(ista-2,0)
+        iend_2u=min(iend+2,im+1)
+      else
+        ista_2l = max(ista - 2,  1 )
+        iend_2u = min(iend + 2, im )
+      endif
 ! special for c-grid v
       jvend_2u = min(jend + 2, jm+1 )
       if(mype==0)print *,'im=',im,'jsta_2l=',jsta_2l,'jend_2u=',jend_2u,'lm=',lm
-!
+      print 901,'GWVX mype/me=',mype,me,'im=',im,'jsta   =',jsta   ,'jend   =',jend   ,'lm=',lm
+      print 901,'GWVX mype/me=',mype,me,'im=',im,'jsta_m =',jsta_m ,'jend_m =',jend_m ,'lm=',lm
+      print 901,'GWVX mype/me=',mype,me,'im=',im,'jsta_2l=',jsta_2l,'jend_2u=',jend_2u,'lm=',lm
+      print 901,'GWVX mype/me=',mype,me,'im=',im,'ista   =',ista   ,'iend   =',iend   ,'lm=',lm
+      print 901,'GWVX mype/me=',mype,me,'im=',im,'ista_m =',ista_m ,'iend_m =',iend_m ,'lm=',lm
+      print 901,'GWVX mype/me=',mype,me,'im=',im,'ista_2l=',ista_2l,'iend_2u=',iend_2u,'lm=',lm
```

Comment:
These print lines will be output from all the write tasks. Please comment them out.

Reply:
Done.